keytree/src/lib.rs

```diff
@@ -37,6 +41,41 @@ impl Xprv {
     }
 }
 
+/// Returns a intermediate child pubkey.
+pub fn intermediate_key(
+    xpub: &Xpub,
+    label: &'static std::string::String,
+    data: &String,
+) -> Option<Xpub> {
+    // question: this function returns an option because it calls a function that returns an option
+    // and I didn't know what else to do. is there something better to do?
+
+    // question: is the following kosher? https://doc.dalek.rs/merlin/struct.Transcript.html#note
+    let mut t = Transcript::new(b"Keytree.intermediate");
+    t.commit_bytes(b"pt", xpub.point.as_bytes());
+    t.commit_bytes(b"dk", &xpub.dk);
+
+    // provide the transcript to the user to commit an arbitrary derivation path or index
+    t.commit_bytes(label.as_bytes(), data.as_bytes());
```

Comment:
Call the closure here:

```suggestion
    customize(&mut t);
```